start_esp/start_esp.py

```diff
@@ -600,6 +605,53 @@ def make_argparser():
     The index usage is the same as the array index in many languages,
     such as Python. This flag is only applied when --client_ip_header is
     specified.''')
+
+    # CORS presets
+    parser.add_argument('--cors_preset',
+        default=None,
+        help='''
+        Intended for enabling CORS with a GRPC backend, since a GRPC
+        backend cannot set CORS headers.
+        Specify one of available presets to configure CORS response headers
+        in nginx. Defaults to no preset and therefore no CORS response
+        headers. If no preset is suitable for the use case, use the
+        --nginx_config arg to use a custom nginx config file.
+        Available presets:
+        - basic - Assumes all location paths have the same CORS policy.
+          Responds to preflight OPTIONS requests with an empty 204, and the
+          results of preflight are allowed to be cached for up to 20 days
+          (1728000 seconds). See descriptions for args --cors_allow_origin,
```

Comment:
Consider making the preflight caching time a customizable parameter, with a default value.